### ISSUE TITLE
Rule Request: Opt-in rule that top-level constants be prefixed by `k`

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#1907]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

**1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.**

Some codebases (i.e. Lyft) consistently require that top-level constants be named `kVarName`.

**2. Provide several examples of what _would_ and _wouldn't_ trigger violations.**

```swift
// Wouldn't trigger
private let kMarginSize = 20.0
public let kFeatureEnabled = false

struct MyStruct {
    let marginSize = 20.0
}

// Would trigger
private let marginSize = 20.0
public let featureEnabled = false
```

**3. Should the rule be configurable, if so what parameters should be configurable?**

We _might_ want to make the prefix configurable (e.g. `k` vs a custom value). I think it'd be fine to do this without allowing configurability for the first version and only add it if people express an interest in it.

**4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.**

Should definitely be opt-in as this is a non-standard stylistic choice.

**Comment by marcelofabri** [comment]
Implemented in #1991.
